### PR TITLE
Continuous builds

### PR DESCRIPTION
**Pull request by dhensby**
Make travis publish our releases on successful merge.

Things that need doing:

1. Turning on travis (make sure you only test pushes and not PRs)
2. Adding an encrypted environment variable GH_TOKEN to allow travis to push to the gh-pages branch

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+sudo: false
+language: php
+php:
+  - '5.6'
+
+env:
+  global:
+      secure: "fQhSvmkxT2mrzIL9tfY54KVRIXWrMZo1bjOx4mfNOrKyrkG3GnIUndSmTAZQU/NTiXDIH+N8aIEP3VpDEfDv4XxI3A+5CAayJKTGuLDcyXy0vVf54BfzebG38oH93v2VE2YTeoG/L+Nb6A4+/hb+/sG2vWol+IKD1av/HxZdZZo="
```
Comment (tractorcow) [inline .travis.yml]: Indentation :D

**Comment by tractorcow** [comment]
> Turning on travis (make sure you only test pushes and not PRs)

I thought that your script does this check? We should always test PRs.

**Comment by tractorcow** [comment]
I'm a bit unsure about "using travis for releases, but can never use it for actual testing"

**Comment by sminnee** [comment]
I'd be more comfortable with this if we had at least a few tests, or if it was building a 'latest' version rather than the mainline version.

**Comment by tractorcow** [comment]
Could we not simply have a script that builds the phar from a local checked out copy, and then ensure via git hooks that any changes to source files also include changes to this binary (or perhaps, if it's possible, actually running this script for us)? Why does this need to be on a separate branch?

**Comment by dhensby** [comment]
I think if we're willing to merge a change (and therefore accept a PR of the corresponding binary) we should be happy auto-building it - this at least stops someone maliciously PR-ing a build to us that actually has some illicit code.

If there's an onus on the merger to test and create the binary and upload it, then we get issues where either the core committer doesn't know this (happened to me) or that it's too much effort and so PRs go un-merged

**Comment by tractorcow** [comment]
> If there's an onus on the merger to test and create the binary and upload it, then we get issues where either the core committer doesn't know this (happened to me) or that it's too much effort and so PRs go un-merged


But we do this already on framework front-end dependencies, and this is arguably a more complicated process.

If it's as simple as running a shell script, wouldn't that be sufficiently low expectations for all contributors to do?

**Comment by dhensby** [comment]
The fact that it needs to be put into another branch and a second PR created is quite a bit more than what we expect in framework/cms

Also, this is a binary that, when submitted as a PR, we have no way of knowing is or isn't correctly compiled or if someone's added malicious code to. 

Using continuous builds means we are protected against that risk as well as reducing the amount of work for a contributor or merger.

**Comment by sminnee** [comment]
OK, I think we can accept the level of risk implied by continuous builds
for the sspak binary, as long as we are careful when merging PRs.

If we start to see broken sspak updates make their way into production, we
will need to prioritise getting a test suite in place.

On Mon, 18 Apr 2016 at 23:02 Daniel Hensby <notifications@github.com> wrote:

> The fact that it needs to be put into another branch and a second PR
> created is quite a bit more than what we expect in framework/cms
>
> Also, this is a binary that, when submitted as a PR, we have no way of
> knowing is or isn't correctly compiled or if someone's added malicious code
> to.
>
> Using continuous builds means we are protected against that risk as well
> as reducing the amount of work for a contributor or merger.
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/silverstripe/sspak/pull/39#issuecomment-211329843>
>
-- 
Sam Minnée
CEO
SilverStripe Limited

**Comment by dhensby** [comment]
OK, great

**Comment by dhensby** [comment]
I'm going to update the secure key and fix the indentation then we can go

**Comment by tractorcow** [comment]
ok, I'm open to the idea of using travis for continuous deployment... have you got a branch that demonstrates this working in action? (on your fork?)

**Comment by tractorcow** [comment]
Ok, let's trial using this.

**Pull request merged**